pkg/operator/customroute/custom_route_controller.go

```diff
@@ -0,0 +1,276 @@
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"github.com/openshift/library-go/pkg/route/routecomponenthelpers"
+)
+
+type customRouteController struct {
+	destSecret     types.NamespacedName
+	componentRoute types.NamespacedName
+	ingressLister  configlistersv1.IngressLister
+	ingressClient  configsetterv1.IngressInterface
+	routeLister    routev1lister.RouteLister
+	routeClient    routeclient.RouteInterface
+	secretLister   corev1listers.SecretLister
+	resourceSyncer resourcesynccontroller.ResourceSyncer
+	operatorClient v1helpers.OperatorClient
+	targetRoute    *routev1.Route
+	consumingUsers []configv1.ConsumingUser
+}
+
+func NewCustomRouteController(
+	componentRouteNamespace string,
+	componentRouteName string,
+	destSecretNamespace string,
+	destSecretName string,
+	targetRoute *routev1.Route,
+	consumingUsers []configv1.ConsumingUser,
+	ingressInformer configinformers.IngressInformer,
+	ingressClient configsetterv1.IngressInterface,
+	routeInformer routeinformer.RouteInformer,
+	routeClient routeclient.RouteInterface,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	operatorClient v1helpers.OperatorClient,
+	eventRecorder events.Recorder,
+	resourceSyncer resourcesynccontroller.ResourceSyncer,
+) factory.Controller {
+	controller := &customRouteController{
+		destSecret: types.NamespacedName{
+			Namespace: destSecretNamespace,
+			Name:      destSecretName,
+		},
+		componentRoute: types.NamespacedName{
+			Namespace: componentRouteNamespace,
+			Name:      componentRouteName,
+		},
+		targetRoute:    targetRoute.DeepCopy(),
+		consumingUsers: consumingUsers,
+		ingressLister:  ingressInformer.Lister(),
+		ingressClient:  ingressClient,
+		routeLister:    routeInformer.Lister(),
+		routeClient:    routeClient,
+		secretLister:   kubeInformersForNamespaces.SecretLister(),
+		operatorClient: operatorClient,
+		resourceSyncer: resourceSyncer,
+	}
+
+	return factory.New().
+		WithInformers(
+			ingressInformer.Informer(),
+			routeInformer.Informer(),
+			kubeInformersForNamespaces.InformersFor("openshift-config").Core().V1().Secrets().Informer(),
+			kubeInformersForNamespaces.InformersFor(destSecretNamespace).Core().V1().Secrets().Informer(),
+		).
+		WithSyncDegradedOnError(operatorClient).
+		WithSync(controller.sync).
+		ResyncEvery(time.Minute).
+		ToController("CustomRouteController", eventRecorder.WithComponentSuffix("custom-route-controller"))
+}
+
+func (c *customRouteController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	ingressConfig, err := c.ingressLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	ingressConfigCopy := ingressConfig.DeepCopy()
+
+	errors := c.syncResources(ctx, syncCtx, ingressConfigCopy)
```

Comment:
nit: the rest of the code usually uses `errs` for error slices

Reply:
Done.